src/Processors/Formats/Impl/NpyRowInputFormat.cpp

```diff
@@ -265,6 +265,11 @@ NpyRowInputFormat::NpyRowInputFormat(ReadBuffer & in_, Block header_, Params par
     nested_type = getNestedType(types[0]);
 }
 
+size_t NpyRowInputFormat::countRows(size_t max_block_size)
+{
+    return int(max_block_size) > header.shape[0] ? header.shape[0] : max_block_size;
```

Comment:
The `max_block_size` means the maximum number of rows we should count and return in a single `countRows` call. It's actually needed to avoid creating huge chunks (I can explain how it works if you are interested). In our case we know the number of rows in advance, but still should consider this limit. So, it can look smth like this:
```c++
size_t count;
if (counted_rows + max_block_size <= header.shape[0])
    count = max_block_size;
else
    count = header.shape[0] - counted_rows;
counted_rows += count;
return count;
```
Where `counted_rows` is a field in NpyRowInputFormat class.